homeassistant/helpers/service.py

```diff
@@ -121,11 +122,23 @@ def extract_entity_ids(hass, service_call, expand_group=True):
 
     entity_ids = await async_extract_entity_ids(hass, service_call, expand_group)
 
-    return [
-        entity
-        for entity in entities
-        if entity.available and entity.entity_id in entity_ids
-    ]
+    found = []
+
+    for entity in entities:
+        if entity.entity_id not in entity_ids:
+            continue
+
+        entity_ids.remove(entity.entity_id)
+
+        if not entity.available:
+            continue
+
+        found.append(entity)
+
+    if entity_ids:
+        raise EntitiesNotFound(entity_ids)
```

Comment:
I would like this. But only in the case of all integration support `unique_id` and are restored on startup. Without this perfect world scenario, you will break automation/action/scripts after startup until everything is set up and the latest retry was successful.